core/app/[locale]/(default)/account/[tab]/_components/wishlist-sheet/index.tsx

```diff
@@ -0,0 +1,70 @@
+import { Heart } from 'lucide-react';
+import { useTranslations } from 'next-intl';
+import { PropsWithChildren, useEffect, useState } from 'react';
+
+import { addWishlistItems } from '~/client/mutations/add-wishlist-items';
+import { Sheet } from '~/components/ui/sheet';
+
+import { Button } from '../../../../../../../components/ui/button';
+
+import { WishlistSheetContent } from './wishlist-sheet-content';
+
+export type Wishlist = NonNullable<Awaited<ReturnType<typeof addWishlistItems>>>;
+
+interface WishlistSheetProps extends PropsWithChildren {
+  productId: number;
+  wishlistsData: Wishlist[];
+}
+
+export const WishlistSheet = ({ productId, wishlistsData }: WishlistSheetProps) => {
+  const t = useTranslations('Account.Wishlist.Sheet');
+
+  const [wishlists, setWishlists] = useState(() => {
+    if (wishlistsData.length === 0) {
+      return [{ items: [], entityId: 0, name: t('favorites') }];
+    }
+
+    return wishlistsData;
+  });
+
+  const [saved, setSaved] = useState(() => {
+    if (wishlistsData.length === 0) {
+      return false;
+    }
+
+    return !!wishlistsData.find((wishlist) => {
+      return !!wishlist.items.find((item) => item.product.entityId === productId);
+    });
```

Comment:
What do you think about using different method on wishlistsData:

```
return wishlistsData.some(({items}) => items.some(({product}) => product.entityId === productId));
```
as I understand we just check if any wish list contains a product with specific Id.

Reply:
I lilke the idea with removing `!!`, thanks